core/src/main/java/org/elasticsearch/index/query/IdsQueryParser.java

```diff
@@ -91,6 +91,8 @@ public IdsQueryBuilder fromXContent(QueryParseContext parseContext) throws IOExc
                 } else {
                     throw new ParsingException(parser.getTokenLocation(), "[ids] query does not support [" + currentFieldName + "]");
                 }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(), "[ids] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?